Trees/BreadthFirstTreeTraversal.js

```diff
@@ -17,7 +17,27 @@ class BinaryTree {
     this.traversal = []
   }
 
-  breadthFirst () {
+  breadthFirstIterative () {
+    this.traversal = []
```

Comment:
I definitely don't like storing the traversal in an instance variable; this should be a local variable and returned (applies for the other traversal methods as well).

Reply:
I agree.

The only reason I can see for doing this was that the original dev was implementing specifically a BFS traversal, so maybe they thought it would be meaningful to store the order. IMO it makes more sense to "do something" at each node, but that's less relevant WRT academia.

Reply:
I was in the process of getting rid of the instance var, but then realized we'd have to alter the original algorithm a bit as well. You'd need to pass in some local variable that gets modified correctly throughout the recursion.

I'm fine to look into this, but can we get the iterative implementation committed first? I can make the other changes we've discussed, but WRT to changing the original algo I'd rather that happen in a separate commit.

Reply:
Sure, we can delay that for now.